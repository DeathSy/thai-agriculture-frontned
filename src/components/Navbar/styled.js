import styled from 'styled-components'

export const Wrapper = styled.div`
  display: flex;
  flex: 1;
  justify-content: space-between;
  width: 100%;
  z-index: 1000;
  background-color: #ffff;
  position: fixed;
  color: #fff;
  box-shadow: 4px 4px 6px 2px rgba(0, 0, 0, 0.2);
  font-family: 'Signika', sans-serif;
`

export const ActionContainer = styled.div`
  width: 50%;
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-items: center;
  margin: 0 2rem;
`
export const Action = styled.div`
  width: 200px;
  height: 50px;
  font-size: 20px;
  border-radius: 50px;
  cursor: pointer;
  border: none;
  background-color: #10c18b;
  color: #fff;
  display: flex;
  justify-content: center;
  align-items: center;
  margin-left: 1rem;
  :hover {
    background-color: #00a472;
  }
`
