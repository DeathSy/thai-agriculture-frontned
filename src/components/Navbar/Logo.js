import React from 'react'
import styled from 'styled-components'

const BackgroundImage = styled.div`
  height: ${props => props.height}vh;
  width: 3vw;
  background-image: url(${props => props.src});
  background-repeat: no-repeat;
  background-size: cover;
  background-position: center;
  position: relative;
  z-index: 11;
  margin-top: 0.5rem;
  margin-bottom:0.5rem;
  margin-left: 1rem;
`
const BackgroundOverylay = styled.div`
  position: absolute;
  width: 100%;
  height: 100%;
  background-color: ${props => (props.isScroll ? '#fff' : 'transparent')};
  color: ${props => (props.isScroll ? '#000' : '#fff')};
  transition: background-color 0.5;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`
function BackgroundBlock ({ src, height = 7, children }) {
  return (
    <div>
      <BackgroundImage src={src} height={height}>
        <BackgroundOverylay>{children}</BackgroundOverylay>
      </BackgroundImage>
    </div>
  )
}
export default BackgroundBlock
