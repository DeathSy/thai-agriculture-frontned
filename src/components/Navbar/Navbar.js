import React from 'react'
import styled from 'styled-components'
import Logo from './Logo'
import Imglogo from '../../assets/logo.png'
import useScroll from '../../util/useScroll'
import { Link } from 'react-router-dom'
import ButtonAddDevice from '../ButtonModal/ButtonDevice'

export const Action = styled.div`
  width: 200px;
  height: 50px;
  font-size: 20px;
  border-radius: 50px;
  cursor: pointer;
  border: none;
  background-color: #10c18b;
  color: #fff;
  display:flex;
  justify-content:center;
  align-items:center;
  margin-left:1rem;
  :hover {
    background-color: #00a472;
  }
`

export const Wrapper = styled.div`
  display: flex;
  flex: 1;
  justify-content: space-between;
  width: 100%;
  z-index: 1000;
  background-color:#FFFF;
  position:fixed;
  color: #fff;
  box-shadow:4px 4px 8px 5px #ccc;
`

export const ActionContainer = styled.div`
  width: 50%;
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-items: center;
  margin: 0 2rem;
`

function Navbar () {
  const isScroll = useScroll({ scrollRange: 20 })
  const i = 0
  if (i === 1) {
    return (
      <Wrapper isScroll={isScroll}>
        <Logo src={Imglogo} />
        <ActionContainer>
          <Action>
            <Link to='/Login'>Login</Link>
          </Action>
          <Action>
            <Link to='/Register'>Register</Link>
          </Action>
        </ActionContainer>
      </Wrapper>
    )
  } else {
    return (
      <Wrapper isScroll={isScroll}>
        <Logo src={Imglogo} />
        <ActionContainer>
          <ButtonAddDevice />
          <Action>Logout</Action>
        </ActionContainer>
      </Wrapper>
    )
  }
}

export default Navbar
