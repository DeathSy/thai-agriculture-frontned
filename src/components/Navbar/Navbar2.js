import React from 'react'
import styled from 'styled-components'
import Logo from './Logo'
import Imglogo from '../../assets/logo.png'

export const Action = styled.div`
  padding: 0.7rem 4rem;
  margin: 0 0.5rem;
  :hover {
    background-color: #00a472;
    border-radius: 60px;
  }
`
export const Wrapper = styled.div`
  display: flex;
  flex: 1;
  align-items: center;
  width: 100%;
  position: fixed;
  background-color: #ffffff;
  z-index: 1000;
`
export const ActionContainer = styled.div`
  display: flex;
  flex: 1;
  justify-content: flex-end;
  align-items: center;
  margin: 0 2rem;
`

function Navbar2 () {
  return (
    <>
      <Wrapper>
        <Logo src={Imglogo} />
        <ActionContainer>
          <Action>
            <>Add Device</>
          </Action>
          <Action>
            <>Logout</>
          </Action>
        </ActionContainer>
      </Wrapper>
    </>
  )
}

export default Navbar2
