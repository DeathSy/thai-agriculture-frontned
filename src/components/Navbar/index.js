import React from 'react'
import { Action, Wrapper, ActionContainer } from './styled'
import Logo from './Logo'
import Imglogo from '../../assets/field.png'
import useScroll from '../../util/useScroll'
import { Link, useHistory } from 'react-router-dom'
import { logout } from '../../services/userAPI'

function Navbar () {
  const history = useHistory()
  const handleClickOut = () => {
    logout(localStorage.getItem('token')).then(() => localStorage.removeItem('token')).then(() => history.push('/'))
  }
  const isScroll = useScroll({ scrollRange: 20 })
  if (!localStorage.getItem('token')) {
    return (
      <Wrapper isScroll={isScroll}>
        <Logo src={Imglogo} />
        <ActionContainer>
          <Action>
            <Link to='/Login'>Login</Link>
          </Action>
          <Action>
            <Link to='/Register'>Register</Link>
          </Action>
        </ActionContainer>
      </Wrapper>
    )
  } else {
    return (
      <Wrapper isScroll={isScroll}>
        <Logo src={Imglogo} />
        <ActionContainer>
          <Action onClick={handleClickOut}>Logout</Action>
        </ActionContainer>
      </Wrapper>
    )
  }
}

export default Navbar
