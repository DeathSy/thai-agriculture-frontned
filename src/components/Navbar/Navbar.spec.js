import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render } from '@testing-library/react'
import Navbar from '.'

test('should render layout correcyly', () => {
  const { Wrapper } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )
  expect(Wrapper).toMatchSnapshot()
})

test('should have link navigate to login', () => {
  const { getByText } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )
  expect(getByText(/login/gi)).toBeInTheDocument()
  expect(getByText(/login/gi).closest('a')).toHaveAttribute('href', '/login')
})

test('should have link navigate to register', () => {
  const { getByText } = render(
    <MemoryRouter>
      <Navbar />
    </MemoryRouter>
  )
  expect(getByText(/register/gi)).toBeInTheDocument()
  expect(getByText(/register/gi).closest('a')).toHaveAttribute('href', '/register')
})