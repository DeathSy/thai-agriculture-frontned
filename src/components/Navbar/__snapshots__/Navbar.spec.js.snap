// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render layout correcyly 1`] = `undefined`;
